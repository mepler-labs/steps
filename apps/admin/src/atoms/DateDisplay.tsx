import moment from 'moment';
import React from 'react';
import { grey } from 'styles/colors';
import styled from 'styled-components';

interface Props {
  date: moment.Moment;
  className?: string;
}

const DateDisplay: React.SFC<Props> = ({ date }) => (
  <StyledDate>{date.format('YYYY-MM-DD')}</StyledDate>
);

const StyledDate = styled.div`
  color ${grey};
`;

export default DateDisplay;
