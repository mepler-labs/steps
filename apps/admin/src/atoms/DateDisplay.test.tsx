import React from 'react';
import moment from 'moment';
import { shallow } from 'enzyme';
import DateDisplay from './DateDisplay';

describe('DateDisplay.tsx', () => {
  const date = moment.utc('2018-01-01');

  it('is defined', () => {
    const wrapper = shallow(<DateDisplay date={date} />);

    expect(wrapper).toBeDefined();
  });

  it('is properly formatted', () => {
    const wrapper = shallow(<DateDisplay date={date} />);

    expect(wrapper.dive().text()).toEqual('2018-01-01');
  });
});
