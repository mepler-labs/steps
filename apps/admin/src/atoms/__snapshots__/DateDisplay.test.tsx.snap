// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateDisplay.tsx is properly formatted 1`] = `
<styled.div>
  2018-01-01
</styled.div>
`;
